### PR TITLE
Extract OIDC user claims in Query Tool

### LINKED ISSUE #1127: Extract OIDC claims/user info for display and logging in the query tool
See #1126 and replicate functionality described there for the dashboard in the query tool.

### PR DESCRIPTION
**Pull request by timoballard**
Closes #1127 

- Extracts the authenticated user's email from OIDC claims and displays in the upper right.
- Adds `RequestLoggingMiddleware`, which logs the authenticated user's email address, request method, and request path for each request.
- Adds `AuthenticationLoggingMiddleware`, which logs all of the authenticated user's claims once when a new session is created.

## Notes
- `IClaimsProvider` is a strongly-typed interface for getting user claims. 
- Claim type mappings are configurable via the [options pattern](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1).
- There are two new `appsettings` files (`appsettings.tts.json` and `appsettings.fns.json`) to allow for configuration of claim types that differ between these environments (e.g., email). The appropriate `appsettings` file is included at app startup automatically (by way of the `ASPNETCORE_ENVIRONMENT` variable) by our use of [CreateDefaultBuilder](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder?view=aspnetcore-3.1) 
- If the `ClaimsProvider` is unable to find an email claim for the current user, most likely due to the claim type being misconfigured, the app currently returns a 500. It might be more appropriate to throw something like a 403 here, as it's an auth/z related issue, but I'd love feedback on [this bit](https://github.com/18F/piipan/blob/a32755a8a1785d08167b0b172f1c0c0eee28ed79/shared/src/Piipan.Shared/ClaimsProvider.cs#L22-L37) in particular!

### CODE REVIEW

File: query-tool/src/Piipan.QueryTool/Program.cs [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.AzureAppServices;
 
 namespace Piipan.QueryTool
 {
+    [ExcludeFromCodeCoverage]
```
Comment (ryanhofdotgov) [inline query-tool/src/Piipan.QueryTool/Program.cs:15]: Probably worth a discussion and ADR on when and where we should do this.

File: tools/build-common.bash [hunk 2]
```diff
@@ -25,7 +25,8 @@ run_tests_ci () {
   dotnet test \
     -p:ContinuousIntegrationBuild=true \
     --collect:"XPlat Code Coverage" \
-    -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=lcov
+    -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=lcov \
+    DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.ExcludeByFile="**/*.cshtml"
```
Comment (ryanhofdotgov) [inline tools/build-common.bash:29]: Can't remember if we've talked about disabling the tests for the templates before – at some point I had hoped we would be able to incorporate something like Cypress (assuming we don't have to support IE11 anymore).

**Review by ryanhofdotgov — COMMENTED** [review]
Nice! I left a few comments, but the .NET 5 vs .NET 3.1 libraries is really the only thing that should be resolved before we can merge this in. Everything else is TBD or a follow-on issue.

**Comment by ryanhofdotgov** [comment]
In our engineering sync this morning, we decided to:
- exclude coverage metrics for start up code, will write up an ADR
- keep including them on the template files (so need to adjust that in this PR), anticipating that we will be adding integration tests that will cover them soon

File: query-tool/src/Piipan.QueryTool/Program.cs [hunk 3]
```diff
@@ -18,6 +22,18 @@ public static void Main(string[] args)
 
         public static IHostBuilder CreateHostBuilder(string[] args) =>
             Host.CreateDefaultBuilder(args)
+                .ConfigureLogging(logging => logging.AddAzureWebAppDiagnostics())
```
[the remark this answers is not in the document]
Comment (timoballard) [reply query-tool/src/Piipan.QueryTool/Program.cs:25]: https://github.com/18F/piipan/issues/1232

File: query-tool/src/Piipan.QueryTool/appsettings.json [hunk 4]
```diff
@@ -6,6 +6,9 @@
       "Microsoft.Hosting.Lifetime": "Information"
     }
   },
+  "Claims": {
```
[the remark this answers is not in the document]
Comment (timoballard) [reply query-tool/src/Piipan.QueryTool/appsettings.json:9]: https://github.com/18F/piipan/issues/1233

File: query-tool/src/Piipan.QueryTool/mock_user.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+[
```
[the remark this answers is not in the document]
Comment (timoballard) [reply query-tool/src/Piipan.QueryTool/mock_user.json:1]: Yes, all still works - excluded!

**Review by ryanhofdotgov — APPROVED** [review]
👍

**Pull request merged**